packages/transformer/src/IModelTransformer.ts

```diff
@@ -1944,6 +1941,31 @@ export class IModelTransformer extends IModelExportHandler {
       ? targetElementId
       : undefined;
 
+    if (this._options.preserveElementIdsForFiltering) {
+      const isInvalid = !Id64.isValid(targetElementId);
+      const sourceInTargetElemProps = this.targetDb.elements.tryGetElementProps(
+        sourceElement.id
+      );
+
+      if (!isInvalid && targetElementId !== sourceElement.id) {
+        // Element found with different id
+        throw new Error(
+          `Element id(${sourceElement.id}) cannot be preserved. Found a different mapping(${targetElementId}) from source element`
+        );
+      }
+      // if we don't find mapping for source element in target(invalid) but another element with source id exists in target
+      else if (isInvalid && sourceInTargetElemProps) {
+        // Element id is already taken by another element
+        throw new Error(
+          `Element id(${sourceElement.id}) cannot be preserved. Found a mapping that is already used by another element(${sourceInTargetElemProps.id}) in target db`
+        );
+      }
+      // Element id in target is available to be remapped
+      else if (isInvalid) {
+        targetElementProps.id = sourceElement.id;
```

Comment:
Not sure what the best way to do this but two options I see:

1)Add optional arg to importElement that tells importer we were in this if statement. ( if we're in this if statement we know we can call insertElement I think? Similarly, if our remapping is not invalid AND the ids are equal then we know we can call update I believe.

2) add public function to importElement: `markElementAsUpdate`. This adds the ID of the element to a set. Importer checks if the ID is in the set, if it is it removes it from the set and calls update. If its not then we know to call insert. 

I sort've lean towards the 2nd option just because the first would be an arg thats specific to preserveElementIdsForFiltering, and Im not sure how it would affect any subclasses of the importer elsewhere, but it does feel more hacky.